README.md

```diff
@@ -1,24 +1,161 @@
+
+
+## Code Implementation
+
+In order to setup a dynamic request feature we will start by writing a test to verify that the page exists (you can ignore the FactoryGirl calls, I'm simply using it to create a new post so we have an object to work with):
+
+```ruby
+# spec/features/post_spec.rb
+
+require 'rails_helper'
+
+feature 'navigate to post pages' do
+  subject { FactoryGirl.create(:post) }
+
+  scenario 'on the show page' do
+    visit "/posts/#{subject.id}"
+    expect(page.status_code).to eq(200)
+  end
+
+end
+```
+
+Running ```bundle exec rspec``` gives us an expected error of: ```ActionController::RoutingError: No route matches [GET] "/posts/1"```. To correct this error let's draw a route that maps to a show action in the posts' controller:
+
+```ruby
+get 'posts/:id', to: 'posts#show'
+```
+
+Here you will notice something that's different from the static route, the ```/:id``` tells the routing system that this route can receive a parameter, and that parameter will be passed to the controller's show action. With this route in place let's run our tests again, you will see we get a different failure this time that says: ```AbstractController::ActionNotFound: The action 'show' could not be found for PostsController```. This means that we need to create a corresponding ```show``` action in the posts' controller, let's get this failure fixed with the code below:
+
+```ruby
+# app/controllers/posts_controller.rb
+
+class PostsController < ApplicationController
+  def show
+  end
+end
+```
+
+Running the tests now you will get a failure saying that that we have a missing view template, let's fix that by creating a view file that is mapped to our ```show``` action by running the following terminal command (we also need to create a posts directory): ```mkdir app/views/posts && touch app/views/posts/show.html.erb```
+
+Running the tests now and we're all green, which means that the request will be properly routes through the controller and view and returns a HTTP status code of ```200```.
+
+Now that we have the routing configured, let's build a test to see if the post content is rendered on the show page with the title being in an ```h1``` tag and the description in a ```p``` tag, below are the scenarios (if you reference the initial spec setup, ```subject``` holds the post that we created with FactoryGirl):
+
+```ruby
+
+# Post factory for reference:
+# title "My Great Post"
+# description "My amazing description"
+
+scenario 'title is shown on the show page in a h1 tag' do
+  visit "/posts/#{subject.id}"
+  expect(page).to have_css("h1", text: "My Great Post")
+end
+
+scenario 'description is shown on the show page in a p tag' do
+  skip
+end
+```
+
+This gives us a failure that says: ```expected to find css "h1" with text "My Great Post" but there were no matches```. To fix this we first need to capture the post's id in the controller's show action, let's do that here:
```

Comment:
I'd change this up a bit to say "we first need to get the `id` sent by the user through the dynamic URL. This variable is passed into the controller in a hash called `params`. Since we named the route `:id`, the id will be the value for the `:id` key. In this case that means it will be in `params[:id]`."